src/main/java/ognl/OgnlRuntime.java

```diff
@@ -3948,12 +3954,9 @@ Method get(Class clazz, String propertyName)
             ConcurrentHashMap<String,Method> methodsByPropertyName = this.cache.get(clazz);
             if (methodsByPropertyName == null)
             {
-                methodsByPropertyName = new ConcurrentHashMap<String, Method>();
-                this.cache.put(clazz, methodsByPropertyName);
+                return null;
             }
             Method method = methodsByPropertyName.get(propertyName);
-            if (method == NULL_REPLACEMENT)
-                return null;
             return method;
```

Comment:
This breaks the logic, you will return `NULL_REPLACEMENT` instead of `null`

Reply:
:+1:

Reply:
What about this? As far I understand this will return `NULL_REPLACEMENT` which was set in [`put`](https://github.com/jkuhnert/ognl/pull/106/files#diff-9464615a5f9bf81dc8354e523e412fa747d62b868f3fbb3270d8a2dd0bda9ee9R3970)

Reply:
Sorry,I can't catch what you said.
Yes, It will return NULL_REPLACEMENT if you put.

Reply:
As far I understand, the previous code will return a raw `null` if `method` from cache equals to `NULL_REPLACEMENT` (which is initialised to `ClassPropertyMethodCache#get`). You've changed the code to return `method` directly without checking if this is `NULL_REPLACEMENT`, so this means you will return `NULL_REPLACEMENT` instead of `null`.

Reply:
It's now checked in the callers (getGetMethod and getSetMethod).

Reply:
Thanks @harawata for clarification :)